### PR TITLE
Feature/analyed based on old files

### PR DESCRIPTION
**Pull request by TheFrok**
I extracted the downloading code to a few function so it will be easier to edit (still not best)
and then added an option to override the date of the source_files so you can now enter any date you would like from the configuration (it sets this date for all the dates in dates_to_analyze)

### CODE REVIEW

File: gtfs/gtfs_utils/gtfs_utils/configuration.py [hunk 1]
```diff
@@ -72,14 +75,27 @@ class Configuration:
...
     display_size_on_progress_bar: bool = True
     delete_downloaded_gtfs_zip_files: bool = True
+    force_existing_files: bool = False
     write_filtered_feed: bool = True
     console_verbosity: str = 'ERROR'
 
 
+def parse_conf_date_format(date_str: str) -> datetime.date:
+    """
+    parse the default date format we use in config.json (YYYY/mm/dd)
```
Comment (cjer) [inline gtfs/gtfs_utils/gtfs_utils/configuration.py:90]: YYYY-mm-dd?

File: gtfs/gtfs_utils/gtfs_utils/gtfs_stats.py [hunk 2]
```diff
@@ -195,6 +178,78 @@ def batch_stats_s3(output_folder: str = None,
...
+    for desired_date in dates_without_output:
+        remote_keys = get_remote_keys_for_date(crud, desired_date)
+        remote_files_mapping[desired_date] = remote_keys
+    return remote_files_mapping
+
+
+def get_remote_keys_for_date(crud, desired_date):
+    """
+    returns a dict mapping an MOT file type to it's newest version before `desired_date`
+    :param crud: S3 Curd object
```
Comment (cjer) [inline gtfs/gtfs_utils/gtfs_utils/gtfs_stats.py:242]: TYPO: S3Crud

**Comment by TheFrok** [comment]
@cjer I fixed the typos

File: gtfs/gtfs_utils/gtfs_utils/s3.py [hunk 3]
```diff
@@ -63,6 +63,14 @@ def get_latest_file(crud: S3Crud,
                     mot_file_name: str,
                     desired_date: Union[datetime.datetime, datetime.date],
                     past_days_to_try: int = 100) -> Tuple[datetime.date, str]:
+    """
+    for mot file type return it's newest version before `desired_date`
```
Comment (OrBin) [inline gtfs/gtfs_utils/gtfs_utils/s3.py:67]: "For MOT file type, return its..."

File: gtfs/gtfs_utils/gtfs_utils/s3.py [hunk 4]
```diff
@@ -63,6 +63,14 @@ def get_latest_file(crud: S3Crud,
...
+    :param crud: S3Crud object
+    :param mot_file_name: Original name of the file from MOT
+    :param desired_date: the newest date acceptable
+    :param past_days_to_try: how back to search for the file
```
Comment (OrBin) [inline gtfs/gtfs_utils/gtfs_utils/s3.py:71]: "how back to search for the file" -> "how many days to search backwards for the file"

**Comment by OrBin** [comment]
@cjer Can you approve?

**Pull request merged**